### PR TITLE
Logs refactoring

### PR DESCRIPTION
**Pull request by zhussupov-t**
Refactored logging: lowered the level for some logs to debag, added instances URI.

### CODE REVIEW

File: internal/vshard/orchestrator/failover.go [hunk 1]
```diff
@@ -131,12 +131,19 @@ func (f *failover) shouldBeAnalysisChecked() bool {
...
 func (f *failover) checkAndRecover(ctx context.Context, analysis *ReplicationAnalysis) {
 	logger := f.logger.With().Str("ReplicaSet", string(analysis.Set.UUID)).Logger()
-	logger.Info().Msgf("checkAndRecover: %s", analysis.String())
+	logger.Debug().Msgf("checkAndRecover: %s", analysis.String())
 
 	recvFunc, desc := f.getCheckAndRecoveryFunc(analysis.State)
 	if recvFunc == nil {
 		if desc != "" {
-			logger.Warn().Msg(desc)
+			master, err := analysis.Set.Master()
```
Comment (pparshin) [inline internal/vshard/orchestrator/failover.go:139]: It seems that you should just add a field `MasterURI` to the `ReplicaSet` and use it.

Comment (zhussupov-t) [reply internal/vshard/orchestrator/failover.go]: Added new field MasterURI

File: internal/vshard/cluster.go [hunk 2]
```diff
@@ -377,6 +379,15 @@ func (c *Cluster) Discover() {
 	c.mutex.Unlock()
 }
 
+func (c *Cluster) getLogLevel(set ReplicaSet) zerolog.Level {
+	previous, err := c.snapshot.ReplicaSet(set.UUID)
+	if err != nil || !previous.SameAs(&set) {
```
Comment (pparshin) [inline internal/vshard/cluster.go:384]: replica set might not be found so it is better to exit quickly:

```go
prev, err := c.snapshot.ReplicaSet(set.UUID)
if err != nil {
  return zerolog.Info
}

changed := !prev.SameAs(set)
if changed {
  return zerolog.Info
}

return zerolog.Debug
```

File: internal/vshard/cluster.go [hunk 3]
```diff
@@ -377,6 +379,15 @@ func (c *Cluster) Discover() {
 	c.mutex.Unlock()
 }
 
+func (c *Cluster) getLogLevel(set ReplicaSet) zerolog.Level {
```
Comment (pparshin) [inline internal/vshard/cluster.go:382]: Create a function `logDiscoveredReplicaSet` and move logging logic inside it

File: internal/vshard/cluster.go [hunk 4]
```diff
@@ -365,8 +367,8 @@ func (c *Cluster) Discover() {
 
 		code, _ := set.HealthStatus()
 		metrics.SetShardCriticalLevel(c.Name, string(set.UUID), int(code))
-
-		c.logger.Debug().Msgf("Discovered: %s", set.String())
+		s := set
```
Comment (pparshin) [inline internal/vshard/cluster.go:370]: Why?

File: internal/vshard/orchestrator/sampler.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	fingerprints map[string]string
+	mu           *sync.RWMutex
+}
+
+func (s *sampler) sample(analysis *ReplicationAnalysis) zerolog.Level {
+	if !s.enabled {
+		return zerolog.InfoLevel
+	}
+
+	s.mu.RLock()
```
Comment (pparshin) [inline internal/vshard/orchestrator/sampler.go:20]: Looks obsolete. What do you protect from here?

File: internal/vshard/replicaset.go [hunk 6]
```diff
@@ -15,11 +16,46 @@ type ReplicaSet struct {
...
+		return false
+	}
+
+	instances := set.Instances
+	anotherInstances := another.Instances
+	sortInstances(instances)
+	sortInstances(anotherInstances)
+
+	for i := 0; i < n; i++ {
+		if instances[i].UUID != anotherInstances[i].UUID ||
```
Comment (pparshin) [inline internal/vshard/replicaset.go:43]: Create the same method `SameAs` for `Instance`

File: internal/vshard/orchestrator/failover.go [hunk 7]
```diff
@@ -130,8 +137,11 @@ func (f *failover) shouldBeAnalysisChecked() bool {
 }
 
 func (f *failover) checkAndRecover(ctx context.Context, analysis *ReplicationAnalysis) {
-	logger := f.logger.With().Str("ReplicaSet", string(analysis.Set.UUID)).Logger()
-	logger.Info().Msgf("checkAndRecover: %s", analysis.String())
+	logger := f.logger.With().
+		Str("ReplicaSet", string(analysis.Set.UUID)).
```
Comment (pparshin) [inline internal/vshard/orchestrator/failover.go:141]: Let's replace UpperCase style to snake_case everywhere in log context.

**Pull request merged**